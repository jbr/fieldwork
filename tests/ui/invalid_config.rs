#[derive(fieldwork::Fieldwork)]
#[fieldwork(invalid_config)]
struct A {
    field: String,
}

#[derive(fieldwork::Fieldwork)]
#[fieldwork(invalid_config = true)]
struct B {
    field: String,
}

#[derive(fieldwork::Fieldwork)]
#[fieldwork(get, set)]
struct C(#[fieldwork(invalid)] String);

#[derive(fieldwork::Fieldwork)]
struct D {
    #[fieldwork(ger)]
    field: (),
}

#[derive(fieldwork::Fieldwork)]
struct E {
    #[fieldwork(get(copy, unknown))]
    field: (),
}

#[derive(fieldwork::Fieldwork)]
#[fieldwork(get(copy, unknown))]
struct F {
    field: (),
}

#[derive(fieldwork::Fieldwork)]
#[fieldwork(bounds = true)]
struct G {
    field: (),
}

#[derive(fieldwork::Fieldwork)]
struct H {
    #[fieldwork(get(rename = true))]
    field: (),
}

#[derive(fieldwork::Fieldwork)]
#[fieldwork(invalid(copy))]
struct I {
    field: (),
}

#[derive(fieldwork::Fieldwork)]
struct J {
    #[fieldwork(get(other = "anything"))]
    field: (),
}

#[derive(fieldwork::Fieldwork)]
struct K {
    #[fieldwork(get(other = anything))]
    field: (),
}

#[derive(fieldwork::Fieldwork)]
#[fieldwork(get(other = "anything"))]
struct L {
    field: (),
}

#[derive(fieldwork::Fieldwork)]
#[fieldwork(other = "anything")]
struct M {
    field: (),
}

#[derive(fieldwork::Fieldwork)]
#[fieldwork((get)(copy))]
struct O(());

#[derive(fieldwork::Fieldwork)]
#[fieldwork((get) = copy)]
struct P(());

#[derive(fieldwork::Fieldwork)]
struct Q(#[fieldwork((get) = copy)] ());

#[derive(fieldwork::Fieldwork)]
struct R(#[fieldwork((get)(copy))] ());

#[derive(fieldwork::Fieldwork)]
struct S(#[fieldwork = 1.0] ());

fn main() {}
