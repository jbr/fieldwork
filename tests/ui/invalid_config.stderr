error: unknown configuration `invalid_config`

       In this position, fieldwork recognizes:
       `chain`, `copy`, `deref`, `into`, `opt_in`, `option`, `option_borrow_inner`,
       `option_set_some`, `rename_predicate`, `rename_predicates`, `vis`, `where_clause`, `bounds`,
       `get`, `set`, `with`, and `get_mut`
 --> tests/ui/invalid_config.rs:2:13
  |
2 | #[fieldwork(invalid_config)]
  |             ^^^^^^^^^^^^^^

error: unknown configuration `invalid_config`

       In this position, fieldwork recognizes:
       `chain`, `copy`, `deref`, `into`, `opt_in`, `option`, `option_borrow_inner`,
       `option_set_some`, `rename_predicate`, `rename_predicates`, `vis`, `where_clause`, `bounds`,
       `get`, `set`, `with`, and `get_mut`
 --> tests/ui/invalid_config.rs:8:13
  |
8 | #[fieldwork(invalid_config = true)]
  |             ^^^^^^^^^^^^^^

error: unknown configuration `invalid`

       In this position, fieldwork recognizes:
       `argument`, `chain`, `copy`, `deref`, `get`, `get_mut`, `into`, `name`, `opt_in`, `option`,
       `option_borrow_inner`, `option_set_some`, `rename`, `rename_predicate`, `set`, `skip`, `vis`,
       and `with`
  --> tests/ui/invalid_config.rs:15:22
   |
15 | struct C(#[fieldwork(invalid)] String);
   |                      ^^^^^^^

error: unknown configuration `ger`; did you mean `get`?

       In this position, fieldwork recognizes:
       `argument`, `chain`, `copy`, `deref`, `get`, `get_mut`, `into`, `name`, `opt_in`, `option`,
       `option_borrow_inner`, `option_set_some`, `rename`, `rename_predicate`, `set`, `skip`, `vis`,
       and `with`
  --> tests/ui/invalid_config.rs:19:17
   |
19 |     #[fieldwork(ger)]
   |                 ^^^

error: unknown configuration `unknown`

       In this position, fieldwork recognizes:
       `argument`, `chain`, `copy`, `deref`, `doc`, `into`, `name`, `opt_in`, `option`,
       `option_borrow_inner`, `option_set_some`, `rename`, `rename_predicate`, `skip`, and `vis`
  --> tests/ui/invalid_config.rs:25:27
   |
25 |     #[fieldwork(get(copy, unknown))]
   |                           ^^^^^^^

error: unknown configuration `unknown`

       In this position, fieldwork recognizes:
       `chain`, `copy`, `deref`, `doc_template`, `into`, `opt_in`, `option`, `option_borrow_inner`,
       `option_set_some`, `rename_predicate`, `rename_predicates`, `skip`, `template`, and `vis`
  --> tests/ui/invalid_config.rs:30:23
   |
30 | #[fieldwork(get(copy, unknown))]
   |                       ^^^^^^^

error: invalid value for key `bounds`
  --> tests/ui/invalid_config.rs:36:13
   |
36 | #[fieldwork(bounds = true)]
   |             ^^^^^^

error: invalid value for key `rename`
  --> tests/ui/invalid_config.rs:43:21
   |
43 |     #[fieldwork(get(rename = true))]
   |                     ^^^^^^
